File: frontend/lib/features/ward/domain/ward.dart
class Shift {
  const Shift({
    required this.id,
    required this.hospital,
    required this.ward,
    required this.specialty,
    required this.shiftType,
    required this.active,
    required this.startedAt,
  });

  final String id;
  final String hospital;
  final String ward;
  final String specialty;
  final String shiftType;
  final bool active;
  final DateTime startedAt;

  factory Shift.fromJson(Map<String, dynamic> json) => Shift(
        id: json['id'] as String,
        hospital: json['hospital'] as String,
        ward: json['ward'] as String,
        specialty: json['specialty'] as String,
        shiftType: json['shift_type'] as String,
        active: json['active'] as bool,
        startedAt: DateTime.parse(json['started_at'] as String),
      );
}

class Patient {
  const Patient({
    required this.id,
    required this.initials,
    required this.age,
    required this.dob,
    required this.sex,
    required this.roomNumber,
    required this.bedNumber,
    required this.diagnosis,
    required this.coMorbids,
    required this.dnar,
    required this.reviewed,
    required this.sortOrder,
    required this.notes,
  });

  final String id;
  final String initials;
  final String age;
  final String dob;
  final String sex;
  final String roomNumber;
  final String bedNumber;
  final String diagnosis;
  final String coMorbids;
  final bool dnar;
  final bool reviewed;
  final int sortOrder;
  final String notes;

  factory Patient.fromJson(Map<String, dynamic> json) => Patient(
        id: json['id'] as String,
        initials: json['initials'] as String,
        age: json['age'] as String,
        dob: json['dob'] as String,
        sex: json['sex'] as String,
        roomNumber: json['room_number'] as String,
        bedNumber: json['bed_number'] as String,
        diagnosis: json['diagnosis'] as String,
        coMorbids: json['co_morbids'] as String,
        dnar: json['dnar'] as bool,
        reviewed: json['reviewed'] as bool,
        sortOrder: json['sort_order'] as int,
        notes: json['notes'] as String,
      );
}

class WardTask {
  const WardTask({
    required this.id,
    required this.patientId,
    required this.title,
    required this.completed,
    required this.priority,
    required this.note,
  });

  final String id;
  final String patientId;
  final String title;
  final bool completed;
  final String priority;
  final String note;

  factory WardTask.fromJson(Map<String, dynamic> json) => WardTask(
        id: json['id'] as String,
        patientId: json['patient_id'] as String,
        title: json['title'] as String,
        completed: json['completed'] as bool,
        priority: json['priority'] as String,
        note: json['note'] as String,
      );
}

File: backend/app/models/ward.py
import uuid
from datetime import datetime

from sqlalchemy import Boolean, DateTime, ForeignKey, Integer, String, Text, func
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column

from app.db.base import Base


class WardShift(Base):
    """Mirrors the legacy's single wc_shift object — one row represents 'the
    current shift'; starting a new one creates a fresh row rather than
    overwriting, so past shifts aren't silently lost, but the app only ever
    reads the most recent row per user (matches legacy behavior exactly)."""

    __tablename__ = "ward_shifts"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False, index=True)
    hospital: Mapped[str] = mapped_column(String(255), nullable=False, default="")
    ward: Mapped[str] = mapped_column(String(255), nullable=False)
    specialty: Mapped[str] = mapped_column(String(255), nullable=False)
    shift_type: Mapped[str] = mapped_column(String(20), nullable=False)  # Morning|Evening|Night|Weekend|On-call
    active: Mapped[bool] = mapped_column(Boolean, nullable=False, default=True)

    started_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())


class WardPatient(Base):
    __tablename__ = "ward_patients"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    user_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("users.id"), nullable=False, index=True)

    initials: Mapped[str] = mapped_column(String(10), nullable=False)
    age: Mapped[str] = mapped_column(String(20), nullable=False, default="")
    dob: Mapped[str] = mapped_column(String(20), nullable=False, default="")
    sex: Mapped[str] = mapped_column(String(10), nullable=False)  # Male|Female|Other
    room_number: Mapped[str] = mapped_column(String(20), nullable=False, default="")
    bed_number: Mapped[str] = mapped_column(String(20), nullable=False, default="")
    diagnosis: Mapped[str] = mapped_column(Text, nullable=False, default="")
    co_morbids: Mapped[str] = mapped_column(Text, nullable=False, default="")
    dnar: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    reviewed: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    sort_order: Mapped[int] = mapped_column(Integer, nullable=False, default=0)
    notes: Mapped[str] = mapped_column(Text, nullable=False, default="")

    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())


class WardTask(Base):
    __tablename__ = "ward_tasks"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    patient_id: Mapped[uuid.UUID] = mapped_column(
        UUID(as_uuid=True), ForeignKey("ward_patients.id", ondelete="CASCADE"), nullable=False, index=True
    )
    title: Mapped[str] = mapped_column(String(500), nullable=False)
    completed: Mapped[bool] = mapped_column(Boolean, nullable=False, default=False)
    priority: Mapped[str] = mapped_column(String(10), nullable=False, default="Medium")  # Low|Medium|High
    note: Mapped[str] = mapped_column(Text, nullable=False, default="")

    created_at: Mapped[datetime] = mapped_column(DateTime(timezone=True), server_default=func.now())
